File: src/Web/Web.Spa/Models/RewardedEmployee.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.ComponentModel.DataAnnotations;
using System.Collections.Generic;

namespace Belorusneft.Museum.Web.Spa.Models
{
    public class RewardedEmployee
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; }

        [Required(ErrorMessage = "Name is not assigned")]
        public string Name { get; set; }

        public string Position { get; set; }

        public DateTime DateBirth { get; set; }

        public DateTime? DateStart { get; set; }

        public DateTime? DateEnd { get; set; }

        public DateTime? DateReward { get; set; }

        public IEnumerable<RewardWithYear> Rewards  { get; set; }

    }
}

File: src/Web/Web.Spa/ClientApp/src/app/models/rewardedemployee.ts
export class RewardedEmployee {
    public name: string;

    public position: string;

    public dateBirth: string;

    public dateStart: string;

    public dateEnd: string;

    public dateReward: string;

    public rewards: {
         rewardId: string;
         dateReward: string;
    }[];
}
